cachi2/core/package_managers/general.py

```diff
@@ -33,6 +44,172 @@ def download_binary_file(url, download_path, auth=None, insecure=False, chunk_si
             f.write(chunk)
 
 
+def clone_repo_pack_archive(
+    url: str,
+    download_dir: Path,
+):
+    """
+    Clone a repository and pack its content as tar using asyncio.
+
+    :param aiohttp_retry.RetryClient session: Aiohttp interface for making HTTP requests.
+    :param str url: URL for file download
+    :param str download_dir: Path to download file to
+    :raise FetchError: If download failed
+    """
+
+    # Parse git URL
+    # Example: git+ssh://git@github.com/kevva/is-positive.git#97edff6f52
+    parse_url = url.replace("git+ssh://", "https://").split("#")
```

Comment:
`Npm install` resolves url with git+ssh and cachi2 fails to clone it.. even though I requested in dependencies `https://bitbucket.org/cachi-testing/cachi2-without-deps-second.git`
See:
* http://pastebin.test.redhat.com/1100437
* https://github.com/cachito-testing/npm-cachi2-smoketest/blob/npm_smoketest_simple/package-lock.json

Reply:
Maybe we have to give up on bitbucket in integration tests then (or modify the lockfiles manually)

I don't think it's acceptable to use a different cloning mechanism than the user intended

Reply:
I am afraid we would need to update lockfiles because I hit the same issue with github resource as well. Are we going to download them based on dependency section? https://github.com/cachito-testing/npm-cachi2-smoketest/blob/npm_smoketest_simple/package-lock.json#L12

Reply:
for now I decided to use callback function